### PR TITLE
Clearer error when show_in_notebook is missing optional dependencies

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address user confusion when `show_in_notebook` now throws ImportError instead of AstropyDeprecationWarning. This is a direct follow-up of https://github.com/astropy/astropy/pull/17165

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/table/table.py [hunk 1]
```diff
@@ -1795,7 +1795,15 @@ def show_in_notebook(self, *, backend="ipydatagrid", **kwargs):
...
         if backend == "ipydatagrid":
-            from astropy.table.notebook_backends import ipydatagrid
+            try:
+                import pandas  # noqa: F401
+
+                from astropy.table.notebook_backends import ipydatagrid
+            except ImportError:
+                raise ImportError(
+                    "The default option for show_in_notebook now requires pandas "
+                    "and ipydatagrid to also be installed"
```
Comment (bsipocz) [inline astropy/table/table.py:1805]:
```suggestion
                    "and ipydatagrid to also be installed, or please consider using the astropy[jupyter] extras"
```

**Review by bsipocz — APPROVED** [review]
Thanks. Please mention the appropriate extras, too, then it should be good to go.

File: astropy/table/table.py (hunk 1 above)
Comment (pllim) [reply astropy/table/table.py:1805]: Hmm. But `pandas` is under `all`, so should the text be `astropy[all,jupyter]`?

Comment (bsipocz) [reply astropy/table/table.py:1805]: If this is a feature that I need for the notebooks then pandas should be in here, too. (And then [jupyter] be included in [all]). Basically the logic should be that the end user shouldn't need to install any of these one by one but by choosing the appropriate extras group(s), but still not being forced to install everything.

Comment (pllim) [reply astropy/table/table.py:1805]: That was done in https://github.com/astropy/astropy/pull/16340 by @nstarman . I did not follow the discussions closely. Reordering pyproject.toml might be out of scope if the ask is only for a clearer error message.

Comment (bsipocz) [reply astropy/table/table.py:1805]: Well, the ask is also to provide an extras that notebooks need, not adding pandas alongside with pydatagrid was imo an oversight.

**Comment by pllim** [comment]
OK I accepted your suggestion and pushed another commit . Please re-review. Thanks!

**Review by bsipocz — APPROVED** [review]
Looks good, thanks for the quick turnaround!

**Pull request merged**